core/cap-0046-01.md

```diff
@@ -181,97 +183,160 @@ See the new XDR files in the Soroban overview CAP.
+  - If both values have an equal bit-pattern, their order is equal.
+  - If _either_ value is an object-handle type, they are compared through object comparison (via the host function `obj_cmp`) as described below.
+  - Otherwise A and B are both small-value types:
+    - If A's `Tag` differs from B's `Tag`, they are ordered by numeric `Tag` value (which, for small values, match the order of the corresponding XDR `SCValType`s).
+    - Otherwise A and B have the same `Tag` value:
+      - If A and B have common tag `Tag::False`, `Tag::True`, `Tag::Void`, or `Tag::LedgerKeyContractInstance`, A and B are equal.
+      - If A and B have common tag `Tag::Error`, A and B are ordered first by their minor components (the "error type"), then by their major components (the "error code"), both treated as unsigned 32-bit integers.
+      - If A and B have common tag `Tag::U32Val`, A and B are ordered by their major components, treated as unsigned 32-bit integers.
+      - If A and B have common tag `Tag::I32Val`, A and B are ordered by their major components, treated as signed 32-bit integers.
+      - If A and B have common tag `Tag::U64Small`, `Tag::U128Small` or `Tag::U256Small`, A and B are ordered by their bodies, treated as unsigned 64-bit integers.
+      - If A and B have common tag `Tag::I64Small`, `Tag::I128Small` or `Tag::I256Small`, A and B are ordered by their bodies, treated as signed 64-bit integers.
+
+Object comparison can be accessed by either guest or host: it is provided to guests as a host function `obj_cmp` via the host environment interface. It performs a recursive structural comparison of objects, as well as values embedded in objects, using the following rules:
+
+  - If either A or B is an invalid handle (does not refer to an object), it is considered less-than a valid object. If both object handles are invalid they are considered equal.
+  - Otherwise both object handles are valid. If A and B have the same `Tag` value, they are directly compared as objects:
+    - If A and B have common tag `Tag::VecObject`, they are ordered by lexicographic extension of the value order.
+    - If A and B have common tag `Tag::MapObject` objects, they are ordered lexicographically as ordered vectors of (key, value) pairs.
+    - If A and B have common tag `Tag::U64Object`, `Tag::I64Object`, `Tag::U128Object`, `Tag::I128Object`, `Tag::U256Object` or `Tag::I256Object`, they are ordered using the numerical order for those types.
+    - If A and B have common tag `Tag::BytesObject`, `Tag::StringObject`, `Tag::SymbolObject`, or `Tag::Address` they are ordered (recursively) in the natural order of their corresponding XDR representations: lexicographically by structure field order, sequence order, union discriminant and structure field numerical orders.
+  - Otherwise only one of A or B are object handles:
+    - If the object handle is invalid (does not refer to an object), it is considered less-than the non-object handle.
```

Comment:
Same here, invalid object handle is no longer allowed.

Reply:
fixed